dev/TODO.txt

```diff
@@ -1,3 +1,15 @@
+- improved random number generation:
+  - swappable generators such as xorshift, pcg32/64, and even just a generic lcg
+  - a good lobster choice may be to abstract them as coroutines, or at least a
+    higher order `def generator(consume):` protocol
```

Comment:
there's currently no way to create a "builtin co-routine" (one written in C++). I'd suggest keeping this simple: just a function that can select which type of rng is currently active.